addons/mass_mailing/static/src/themes/theme_selector/theme_selector.js

```diff
@@ -35,6 +41,19 @@ export class ThemeSelector extends Component {
                 }))
             );
         });
+        onMounted(async () => {
+            for (const template of this.favoriteTemplates) {
+                const iframe = document.querySelector("#iframe_fav_" + template.id);
+                if (iframe.contentDocument.readyState === "complete") {
+                    await this.setupIframe(template, iframe);
```

Comment:
You could regroup those promises into an array that you can Promise.all after your loop. This way we don't wait for each iframes and have a bit better performance. #everyMsCounts